clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -205,14 +211,24 @@ public abstract class AbstractMembershipManager<R extends AbstractResponse> impl
      */
     private boolean isPollTimerExpired;
 
+    /**
+     * Indicate the operation on consumer group membership that the consumer will perform when leaving the group.
+     * The property should remain {@code GroupMembershipOperation.DEFAULT} until the consumer is closing.
+     *
+     * @see GroupMembershipOperation
+     */
+    protected GroupMembershipOperation leaveGroupOperation = GroupMembershipOperation.DEFAULT;
```

Comment:
Due to the architecture of the async consumer, it's difficult to pass `GroupMembershipOperation` directly to `AbstractMembershipManager`, so I added a property to hold the behavior for leaving the group, which remains unchanged except when the consumer is closing.